src/main/java/bio/terra/service/dataset/StorageResource.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.service.dataset;
+
+import bio.terra.model.CloudPlatform;
+
+public class StorageResource {
+
+    private CloudPlatform cloudPlatform;
+    private String cloudResource;
```

Comment:
Would it be possible to make this field `GoogleCloudResource` instead of `String`? That could avoid some enum-to-string conversions done elsewhere in this PR.